File: backend/src/main/java/com/discipolat/modules/groupMessages/domain/GroupMessage.java
package com.discipolat.modules.groupMessages.domain;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "group_messages")
@org.hibernate.annotations.FilterDef(name = "tenantFilter", parameters = @org.hibernate.annotations.ParamDef(name = "tenantId", type = UUID.class))
@org.hibernate.annotations.Filter(name = "tenantFilter", condition = "tenant_id = :tenantId")
public class GroupMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private UUID tenantId;

    @Column(nullable = false)
    private UUID groupId; // department or family id

    @Column(nullable = false)
    private String groupType; // DEPARTMENT, FAMILY, TEAM

    @Column(nullable = false)
    private UUID senderId;

    @Column(columnDefinition = "TEXT")
    private String content;

    private String messageType = "TEXT"; // TEXT, IMAGE, FILE, VOICE, REACTION

    private String fileName;

    private String replyToId;

    private Integer reactionCount = 0;

    @Column(nullable = false)
    private Boolean isDeleted = false;

    private LocalDateTime createdAt = LocalDateTime.now();

    // Getters & setters
    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }
    public UUID getTenantId() { return tenantId; }
    public void setTenantId(UUID tenantId) { this.tenantId = tenantId; }
    public UUID getGroupId() { return groupId; }
    public void setGroupId(UUID groupId) { this.groupId = groupId; }
    public String getGroupType() { return groupType; }
    public void setGroupType(String groupType) { this.groupType = groupType; }
    public UUID getSenderId() { return senderId; }
    public void setSenderId(UUID senderId) { this.senderId = senderId; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getMessageType() { return messageType; }
    public void setMessageType(String messageType) { this.messageType = messageType; }
    public String getFileName() { return fileName; }
    public void setFileName(String fileName) { this.fileName = fileName; }
    public String getReplyToId() { return replyToId; }
    public void setReplyToId(String replyToId) { this.replyToId = replyToId; }
    public Integer getReactionCount() { return reactionCount; }
    public void setReactionCount(Integer reactionCount) { this.reactionCount = reactionCount; }
    public Boolean getIsDeleted() { return isDeleted; }
    public void setIsDeleted(Boolean isDeleted) { this.isDeleted = isDeleted; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}

File: mobile/lib/models/group_message.dart
/// P10 — Modèle message de groupe (mobile).
///
/// Correspond au payload JSON de `GroupMessageController` (entity
/// `com.discipolat.modules.groupMessages.domain.GroupMessage`).
class GroupMessage {
  final String id;
  final String groupId;
  final String groupType;
  final String senderId;
  final String content;
  final String messageType;
  final int reactionCount;
  final bool isDeleted;
  final DateTime createdAt;

  GroupMessage({
    required this.id,
    required this.groupId,
    required this.groupType,
    required this.senderId,
    required this.content,
    required this.messageType,
    required this.reactionCount,
    required this.isDeleted,
    required this.createdAt,
  });

  factory GroupMessage.fromJson(Map<String, dynamic> json) => GroupMessage(
        id: (json['id'] ?? json['Id'])?.toString() ?? '',
        groupId: (json['groupId'] ?? json['groupId']?.toString()) ?? '',
        groupType: json['groupType']?.toString() ?? 'DEPARTMENT',
        senderId: json['senderId']?.toString() ?? '',
        content: json['content']?.toString() ?? '',
        messageType: json['messageType']?.toString() ?? 'TEXT',
        reactionCount: json['reactionCount'] is int
            ? json['reactionCount'] as int
            : int.tryParse(json['reactionCount']?.toString() ?? '0') ?? 0,
        isDeleted: json['isDeleted'] is bool
            ? json['isDeleted'] as bool
            : (json['isDeleted']?.toString().toLowerCase() == 'true'),
        createdAt: DateTime.tryParse(json['createdAt']?.toString() ?? '') ??
            DateTime.now(),
      );
}
